src/passes/Asyncify.cpp

```diff
@@ -1301,4 +1306,138 @@ struct Asyncify : public Pass {
 
 Pass* createAsyncifyPass() { return new Asyncify(); }
 
+// Helper passes that can be run after Asyncify.
+
+template<bool neverRewind, bool neverUnwind, bool importsAlwaysUnwind>
+struct PostAsyncify
```

Comment:
If I'm understanding this at a high level, these are passes that run on already-asyncified code, in order to bake in assumptions about when the code is called, so we can optimize the runtime overhead away.

I'm not sure if "post asyncify" is a great name for this pass. That just says "this runs afterward" but doesn't explain why, or what it does. In particular it seems reasonable to assume that PostAsyncify always runs and does some sort of necessary fixup, rather than optimizing.
So maybe "optimize asyncify"? That's not quite right either because it misses the connotation of baking in assumptions.

Similarly, there's three bools here, but we only instantiate it two ways. Another way to factor this is as explicitly supporting the lazy-load case, and then the pass can be called "lazy load asyncify", parameterized on whether this is the loader or the loadee. If this is expected to be more general, and have options or configurations added, it would make sense to go the other way and have this take an options struct (or bitfield of flags) to keep the parameter list shorter.

Reply:
Yeah, that's a good point, "post" is not clear enough.

I think this is more general than lazy loading, though, so I'd prefer something more general somehow.

Maybe `PostAsyncify => `AsyncifyAssume` because it bakes assumptions into asyncified code?